scripts/cobot280pi/server.py

```diff
@@ -0,0 +1,174 @@
+		try:
+			self.mc.set_gripper_value(**args)
+			time.sleep(3)
+			return {"success": True}
+		except Exception as e:
+			logger.critical("control gripper error: " + truncate_string(str(e)))
+			return {"success": False, "error_msg": str(e)}
+
+	def handle_control_angle(self, payload):
+		args = payload["args"]
+		try:
+			self.mc.send_angles(**args)
+			time.sleep(3)
+			return {"success": True}
+		except Exception as e:
+			logger.critical("control angle error: " + truncate_string(str(e)))
+			return {"success": False, "error_msg": str(e)}
+
+	def handle_control_coord(self, payload):
+		args = payload["args"]
+		try:
+			self.mc.send_coords(**args)
+			time.sleep(3)
+			return {"success": True}
+		except Exception as e:
+			logger.critical("control coords error: " + truncate_string(str(e)))
+			return {"success": False, "error_msg": str(e)}
+
+	def handle_query_angle(self, payload):
+		args = payload["args"]
+		try:
+			angles = self.mc.get_angles()
+			time.sleep(3)
+			if angles is None or len(angles) < 6:
+				raise Exception("could not read angle")
+			return {"success": True, "angles": angles}
+		except Exception as e:
+			error_msg = str(e)
+			logger.critical(error_msg)
+			return {"success": False, "error_msg": truncate_string(error_msg)}
+
+	def handle_query_coord(self, payload):
+		args = payload["args"]
+		try:
+			coords = self.mc.get_coords()
+			time.sleep(3)
+			if coords is None or len(coords) < 6:
+				raise Exception("could not read coord")
+			return {"success": True, "coords": coords}
+		except Exception as e:
+			error_msg = str(e)
+			logger.critical(error_msg)
+			return {"success": False, "error_msg": truncate_string(error_msg)}
+
+	# TODO: fix bug here
+	def handle_query_camera(self, payload):
+		args = payload["args"]
+		try:
+			webcam = cv2.VideoCapture(0)
+			_, frame = webcam.read()
+			frame_arr = np.array(frame)
+			pixel_list = frame_arr.astype(int).flatten().tolist()
+			webcam.release()
+			return {
+				"success": True,
+				"image_pixels": pixel_list,
+    			"image_shape": list(frame_arr.shape),
+				"image_type": frame_arr.dtype.name
+			}
+		except Exception as e:
+			logger.critical(f"error in camera query: {truncate_string(str(e))}")
+			return {"success": False, "error_msg": str(e)}
+			
+
+
+def on_connect(client, userdata, flags, rc, properties=None):
+	print("CONNACK received with code %s." % rc)
+
+def on_publish(client, userdata, mid, properties=None):
+	print("mid: " + str(mid))
+
+def on_message(client, userdata, msg):
+	print(msg.topic + " " + str(msg.qos) + " " + str(msg.payload))
+	handler.push_command(msg)
+
+
+if __name__ == "__main__":
+	client = paho.Client(client_id="", userdata=None, protocol=paho.MQTTv5)
+	client.on_connect = on_connect
+	client.on_message = on_message
+	client.on_publish = on_publish
+
+	client.tls_set(tls_version=paho.ssl.PROTOCOL_TLS)
+	client.username_pw_set(HIVEMQ_USERNAME, HIVEMQ_PASSWORD)
+	client.connect(HIVEMQ_HOST, 8883)
+
+	client.subscribe("#", qos=2)

```

Comment:
Worth thinking about what the schema will be. Maybe prepended by `cobot280pi/<pi-device-id>`